### PR TITLE
server: set timeout for MoveLeader

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve? 
When multiple PDs calls `MoveLeader`, one will succeed and others will block in `MoveLeader` function.

### What is changed and how it works?
Add timeout when calling `MoveLeader`.

### Check List 

Tests 
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release notes

### CODE REVIEW

File: server/leader.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ import (
 	"go.uber.org/zap"
 )
 
+// The timeout to wait transfer etcd leader to complete.
+const moveLeaderTimeout = 5 * time.Second
```
Comment (siddontang) [inline server/leader.go]: can it be configured?

Comment (disksing) [reply server/leader.go]: No. However, I think 5 seconds is a reasonable value and users are unlikely to need to modify it.

**Comment by disksing** [comment]
port in use.
/rebuild

**Pull request merged**